bindings/gumjs/gumv8memory.cpp

```diff
@@ -349,6 +351,29 @@ GUMJS_DEFINE_FUNCTION (gumjs_memory_protect)
   info.GetReturnValue ().Set (success);
 }
 
+GUMJS_DEFINE_FUNCTION (gumjs_memory_query_protection)
+{
+  gpointer address;
+  GumPageProtection prot;
+
+  if (!_gum_v8_args_parse (args, "p", &address))
+    return;
+
+  if (!gum_memory_query_protection (address, &prot))
+    return;
+
+  gchar prot_str[4] = "---";
+
+  if ((prot & GUM_PAGE_READ) != 0)
+    prot_str[0] = 'r';
+  if ((prot & GUM_PAGE_WRITE) != 0)
+    prot_str[1] = 'w';
+  if ((prot & GUM_PAGE_EXECUTE) != 0)
+    prot_str[2] = 'x';
+
+  info.GetReturnValue ().Set (_gum_v8_string_new_ascii (isolate, prot_str));
```

Comment:
Would be good to avoid duplicating this. Perhaps something like:

```diff
diff --git a/bindings/gumjs/gumv8value.h b/bindings/gumjs/gumv8value.h
index 999d1952..3e72f01d 100644
--- a/bindings/gumjs/gumv8value.h
+++ b/bindings/gumjs/gumv8value.h
@@ -162,6 +162,8 @@ G_GNUC_INTERNAL GArray * _gum_v8_memory_ranges_get (v8::Local<v8::Value> value,
 G_GNUC_INTERNAL gboolean _gum_v8_memory_range_get (v8::Local<v8::Value> value,
     GumMemoryRange * range, GumV8Core * core);

+G_GNUC_INTERNAL v8::Local<v8::String> _gum_v8_page_protection_new (
+    v8::Isolate * isolate, GumPageProtection prot);
 G_GNUC_INTERNAL gboolean _gum_v8_page_protection_get (
     v8::Local<v8::Value> prot_val, GumPageProtection * prot,
     GumV8Core * core);
```